packages/react/__tests__/get-label-from-stack-trace.js

```diff
@@ -0,0 +1,560 @@
+    at eval (webpack-internal:///./pages/index.js:78:99)
+    at Array.map (<anonymous>)
+    at MyComponent$9 (webpack-internal:///./pages/index.js:78:19)
+    at processChild (C:/Projects/OSS/emotion/node_modules/react-dom/cjs/react-dom-server.node.development.js:3043:14)
+    at resolve (C:/Projects/OSS/emotion/node_modules/react-dom/cjs/react-dom-server.node.development.js:2960:5)
+    at ReactDOMServerRenderer.render (C:/Projects/OSS/emotion/node_modules/react-dom/cjs/react-dom-server.node.development.js:3435:22)
+    at ReactDOMServerRenderer.read (C:/Projects/OSS/emotion/node_modules/react-dom/cjs/react-dom-server.node.development.js:3373:29)
+    at Object.renderToString (C:/Projects/OSS/emotion/node_modules/react-dom/cjs/react-dom-server.node.development.js:3988:27)`
+
+    expect(getLabelFromStackTrace(stackTrace)).toBe(expectedLabel)
+  })
+})
+
+/**
+ * E.g.
+ *
+ * ```
+ * const test = {
+ *   MyComponent$9() {
+ *     return <div css={{ color: 'red' }}>red</div>
+ *   }
+ * }
+ *
+ * function App() {
+ *   const el = test.MyComponent$9()
+ *
+ *   return ...
+ * }
+ * ```
+ */
+describe('function component within object', () => {
+  test('Chrome', () => {
+    const stackTrace = `Error
+    at createEmotionProps (emotion-element-6352414e.browser.esm.js?fcc6:142)
+    at jsxDEV (emotion-react-jsx-dev-runtime.browser.esm.js?cf67:18)
+    at Object.MyComponent$9 (index.js?bee7:7)
+    at App (index.js?bee7:14)
+    at renderWithHooks (react-dom.development.js?3c4a:14803)
+    at mountIndeterminateComponent (react-dom.development.js?3c4a:17482)
+    at beginWork (react-dom.development.js?3c4a:18596)
+    at beginWork$1 (react-dom.development.js?3c4a:23179)`
+
+    expect(getLabelFromStackTrace(stackTrace)).toBe(expectedLabel)
+  })
+
+  test('Firefox', () => {
+    const stackTrace = `createEmotionProps@webpack-internal:///../../packages/react/dist/emotion-element-6352414e.browser.esm.js:163:40
+jsxDEV@webpack-internal:///../../packages/react/jsx-dev-runtime/dist/emotion-react-jsx-dev-runtime.browser.esm.js:35:230
+MyComponent$9@webpack-internal:///./pages/index.js:32:82
+App@webpack-internal:///./pages/index.js:49:17
+renderWithHooks@webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:14803:27
+mountIndeterminateComponent@webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:17482:13
+beginWork@webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:18596:16
+beginWork$1@webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:23179:14`
+
+    expect(getLabelFromStackTrace(stackTrace)).toBe(expectedLabel)
+  })
+
+  test('Safari', () => {
+    // MyComponent$9 does not appear in the stack trace
+    const stackTrace = `createEmotionProps@http://localhost:3000/static/js/main.chunk.js:866:49
+jsxDEV@http://localhost:3000/static/js/main.chunk.js:1147:247
+App@http://localhost:3000/static/js/main.chunk.js:2290:32
```

Comment:
this is gonna be hard(er) to handle in the potential console.warn/error override - since in here we extract the label, it's just a wrong one

Reply:
Agreed. I _think_ I will still be able to suppress the warning in this case...